rpc_util.go

```diff
@@ -231,6 +239,48 @@ func PerRPCCredentials(creds credentials.PerRPCCredentials) CallOption {
 	})
 }
 
+// CallContentSubtype returns a CallOption that will set the content-subtype
+// for a call. For example, if content-subtype is "json", the Content-Type
+// over the wire will be "application/grpc+json". See Content-Type on
+// https://grpc.io/docs/guides/wire.html#requests for more details. Note that
+// content-subtypes are case-insensitive.
+//
+// If CallCustomCodec is not also used, the content-subtype will be used to
+// look up the Codec to use in the registry controlled by RegisterCodec. See
+// the documention on RegisterCodec for details on registration. If no such
+// Codec is found, the call will result in an error with code codes.Internal.
+//
+// If CallCustomCodec is used, this Codec will be used for all request and
+// response messages, with the content-subtype still being set to the given
+// content-subtype for all requests.
+func CallContentSubtype(contentSubtype string) CallOption {
+	contentSubtype = strings.ToLower(contentSubtype)
+	return beforeCall(func(c *callInfo) error {
+		if contentSubtype == "" {
+			return errEmptyContentSubtypeCallContentSubtype
```

Comment:
I would prefer to ignore it (just send "application/grpc") when this happens.

Reply:
I think we should explicitly error - if you use the `CallContentSubtype` option, passing an empty string ie `CallContentSubtype("")` should be a bad input, defaulting to "application/grpc" seems like undefined behavior to me, right?

Reply:
Undefined just means we haven't defined it.  :)  I'd rather explicitly define it that way.  Then users can do:
```go
cc := grpc.Dial(..., grpc.WithDefaultCallOptions(grpc.CallContentSubtype("blah")))
...
cc.Call(..., grpc.CallContentSubtype(""))
```
To restore the default behavior for that one call.